### PR TITLE
Add pt2e dynamic quantization

### PR DESCRIPTION
**Pull request by yiliu30**
## Type of Change

feature
API changed or not

### Usage

```python
# User script
model = UserModel()
example_inputs = ...

# quantize script

# import intel_extension_for_pytorch ### <--- if user want to use ipex' static quant

from neural_compressor.torch.quantization import get_default_dynamic_config, prepare, convert
from neural_compressor.torch.export import export

# export
dynamic_shapes = {"input_ids": (None, Dim("seq_len"))}
exported_model = export(model, example_inputs=example_inputs, dynamic_shapes=dynamic_shapes)

# prepare
quant_config = get_default_dynamic_config()
prepare_model = prepare(model, quant_config)

# calibrate
run_fn(prepare_model)

# convert
converted_model = convert(prepare_model)

# compile and inference
opt_model = torch.compile(converted_model)
out = opt_model(*example_inputs)
```

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/torch/quantization/config.py [hunk 1]
```diff
@@ -778,6 +779,80 @@ def get_default_AutoRound_config() -> AutoRoundConfig:
     return AutoRoundConfig()
 
 
+######################## Static Quant Config ###############################
```
Comment (yuwenzho) [inline neural_compressor/torch/quantization/config.py:782]: should be "Dynamic Quant Config"

File: neural_compressor/torch/utils/constants.py [hunk 2]
```diff
@@ -52,3 +52,4 @@
 
 
 PT2E_STATIC_QUANT = "pt2e_static_quant"
+PT2E_DYNAMIC_QUANT = "pt2e_dynamic_quant"
```
Comment (yuwenzho) [inline neural_compressor/torch/utils/constants.py:55]: Seems there is no config registered with PT2E_STATIC_QUANT in config.py. How `config` is mapped to `pt2e_static_quant_entry`?

File: neural_compressor/torch/quantization/config.py (hunk 1 above)
Comment (yiliu30) [reply neural_compressor/torch/quantization/config.py]: Thanks, corrected.

File: neural_compressor/torch/utils/constants.py (hunk 2 above)
Comment (yiliu30) [reply neural_compressor/torch/utils/constants.py:55]: Yes, it's a little trick, as we do not want expose two static quant configs to users. 
At `static_quant_entry`, we redispatch the config to `pt2e_static_quant_entry` if ipex if not imported.

https://github.com/intel/neural-compressor/blob/d4b0f0f1e6d85345c3c7eaee5abba0fb904aa4cc/neural_compressor/torch/quantization/algorithm_entry.py#L143-L151

File: neural_compressor/torch/utils/environ.py [hunk 3]
```diff
@@ -31,20 +31,20 @@ def is_hpex_available():
...
 
-try:
-    import intel_extension_for_pytorch as ipex
-
-    _ipex_available = True
-except:
-    _ipex_available = False
-
-
 def is_ipex_available():
```
Comment (xin3he) [inline neural_compressor/torch/utils/environ.py:34]: please don't wrap it in function, we should make env check once for all.

**Review by xin3he — CHANGES_REQUESTED** [review]

**Review by xin3he — APPROVED** [review]

**Review by Kaihui-intel — APPROVED** [review]

**Review by yuwenzho — APPROVED** [review]

File: neural_compressor/torch/utils/environ.py (hunk 3 above)
Comment (yiliu30) [reply neural_compressor/torch/utils/environ.py:34]: Keep it under the function, since we do not want to import ipex automatically.

**Pull request merged**